### PR TITLE
[CI:DOCS]podman-machine and init documentation cleanups

### PR DESCRIPTION
**Pull request by baude**
small tweaks and fixes to reflect more more accurately how Podman machine works.

[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: docs/source/markdown/podman-machine-init.1.md.in [hunk 1]
```diff
@@ -28,27 +28,6 @@ By default, the VM distribution is [Fedora CoreOS](https://getfedora.org/en/core
...
-machine.  Executing `podman info` should reveal versions of both.  A configuration where the Podman host and machine
-mismatch are unsupported.
-
-For more information on updates and advanced configuration,
-see the Fedora CoreOS documentation about [auto-updates](https://docs.fedoraproject.org/en-US/fedora-coreos/auto-updates/) and [update strategies](https://coreos.github.io/zincati/usage/updates-strategy/).
-
-
-
-
-Fedora CoreOS upgrades come out every 14 days and are detected and installed automatically. The VM is rebooted during the upgrade.
```
Comment (Luap99) [inline docs/source/markdown/podman-machine-init.1.md.in:51]: why is this removed?

File: docs/source/markdown/podman-machine.1.md [hunk 2]
```diff
@@ -1,42 +1,42 @@
...
+
+## DESCRIPTION
+`podman machine` is a set of subcommands that manage Podman's virtual machine.
+
+Podman on MacOS and Windows requires a virtual machine. This is because Podman runs Linux containers -
+Linux containers do not run on any other OS because containers' core functionality are
+tied to the Linux kernel. Podman machine must be used to manage MacOS and Windows machines,
+but can be optionally used on Linux.
+
+If you are running `podman machine` in Linux, all `podman machine` commands are rootless only.
```
Comment (Luap99) [inline docs/source/markdown/podman-machine.1.md:17]: I think @TomSweeneyRedHat does not like `you` in man pages.

File: docs/source/markdown/podman-machine-init.1.md.in (hunk 1 above)
Comment (baude) [reply docs/source/markdown/podman-machine-init.1.md.in:51]: it is not accurate ... we do no encourage updates to be run like this.  Moreover, in Podman 5, this will be completely wrong.

File: docs/source/markdown/podman-machine.1.md (hunk 2 above)
Comment (rhatdan) [reply docs/source/markdown/podman-machine.1.md:17]: When running ...
